src/plugins/mf_priority.cpp

```diff
@@ -1116,6 +1144,42 @@ extern "C" int flux_plugin_init (flux_plugin_t *p)
     priority_weights["queue"] = DEFAULT_QUEUE_WEIGHT;
     priority_weights["age"] = DEFAULT_AGE_WEIGHT;
 
+    // initialize the plugin with total node and core counts
+    flux_t *h;
+    flux_future_t *f;
+    char *core;
+
+    h = flux_jobtap_get_flux (p);
+    if (!(f = flux_kvs_lookup (h,
+                               NULL,
+                               FLUX_KVS_WATCH | FLUX_KVS_WAITCREATE,
+                               "resource.R"))) {
+        flux_log_error (h, "flux_kvs_lookup");
+        return -1;
+    }
+
+    if (flux_kvs_lookup_get_unpack (f,
+                                    "{s{s[{s{s:s}}]}}",
+                                    "execution",
+                                      "R_lite",
+                                        "children",
+                                          "core", &core) < 0) {
+        flux_log_error (h, "flux_kvs_lookup_unpack");
+        return -1;
+    }
+
+    if (core == NULL) {
+        flux_log_error (h,
+                        "mf_priority: could not get system "
+                        "cores-per-node information");
+        return -1;
+    }
+
+    // calculate number of cores-per-node on system
+    ncores_per_node = calculate_range (core);
```

Comment:
Instead of parsing the idset with custom code here, you can use libflux-idset.so exported by flux-core.
See [idset_decode(3)](https://flux-framework.readthedocs.io/projects/flux-core/en/stable/man3/idset_decode.html), then just use [idset_count(3)](https://flux-framework.readthedocs.io/projects/flux-core/en/stable/man3/idset_create.html) to get the number of cores.